src/main/java/io/github/thebusybiscuit/slimefun4/implementation/accessibility/PlayerAccessManagerImpl.java

```diff
@@ -0,0 +1,247 @@
+        }
+        return Optional.empty();
+    }
+
+    @Nonnull
+    @Override
+    public <T> AccessData<T> getOrRegisterAccessData(@Nonnull final Class<T> clazz,
+                                                     @Nonnull final AccessData<T> def) {
+        return getAccessData(clazz).orElseGet(() -> {
+            dataMap.put(clazz, def);
+            return def;
+        });
+    }
+
+    @Override
+    public boolean hasDataFor(@Nonnull final Object object) {
+        if (object instanceof Player) {
+            return getAccessData(Player.class).map(data -> data.hasDataFor((Player) object))
+                .orElse(false);
+        } else if (object instanceof UUID) {
+            return getAccessData(UUID.class).map(data -> data.hasDataFor((UUID) object))
+                .orElse(false);
+        }
+        final Optional<? extends AccessData> optional = getAccessData(object.getClass());
+        return optional.map(accessData -> accessData.hasDataFor(object)).orElse(false);
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return dataMap.isEmpty();
+    }
+
+    @Override
+    public void reset() {
+        this.dataMap.clear();
+    }
+
+    @Override
+    public void load(@Nonnull NamespacedKey namespace, @Nonnull ItemStack itemStack) {
+        reset();
+        if (!SlimefunPlugin.getMinecraftVersion().isAtLeast(MinecraftVersion.MINECRAFT_1_14)) {
+            // This feature was introduced post 1.13. Cannot load newer data on an older version
+            return;
+        }
+        final ItemMeta meta = itemStack.getItemMeta();
+        if (meta != null) {
+            load(namespace, meta);
+        }
+
+    }
+
+    @Override
+    public void load(@Nonnull final NamespacedKey namespace, @Nonnull final Block block) {
+        reset();
+        if (!SlimefunPlugin.getMinecraftVersion().isAtLeast(MinecraftVersion.MINECRAFT_1_14)) {
+            return;
+        }
+        final BlockState state = block.getState();
+        if (state instanceof TileState) {
+            load(namespace, ((TileState) state));
+        }
+    }
+
+    @Override
+    public void saveTo(@Nonnull NamespacedKey namespace, @Nonnull final ItemStack item) {
+        if (!SlimefunPlugin.getMinecraftVersion().isAtLeast(MinecraftVersion.MINECRAFT_1_14)) {
+            // This feature was introduced post 1.13. Cannot save.
+            return;
+        }
+        final ItemMeta meta = item.getItemMeta();
+        if (meta != null) {
+            saveTo(namespace, meta);
+            item.setItemMeta(meta);
+        }
+    }
+
+    @Override
+    public void saveTo(@Nonnull final NamespacedKey namespace, @Nonnull final Block block) {
+        if (!SlimefunPlugin.getMinecraftVersion().isAtLeast(MinecraftVersion.MINECRAFT_1_14)) {
+            return;
+        }
+        final BlockState state = block.getState();
+        if (state instanceof TileState) {
+            saveTo(namespace, ((TileState) state));
+            state.update(true, false);
```

Comment:
You only need to update snapshots, so you'll need to check if it is a snapshot.

Reply:
How would I go about doing this? Or did you mean whether the tile state is "virtual"? (It shouldn't be since I've gotten it off a block).